SimplePolicy.UnitTests/UT_SimplePolicy.cs

```diff
@@ -22,5 +30,225 @@ public void TestMaxTransactionsPerBlock()
             Settings.Default.MaxTransactionsPerBlock.Should().Be(500);
             Settings.Default.MaxFreeTransactionsPerBlock.Should().Be(20);
         }
+
+        [TestMethod]
+        public void TestFilterForBlock_ClaimHasPriority()
+        {
+            // Should contain "ClaimTransaction" in "HighPriorityTxType"
+            Settings.Default.HighPriorityTxType.Contains(TransactionType.ClaimTransaction).Should().Be(true);
+
+            ClaimTransaction claimTxZero = GetClaimTransaction(0);
+            claimTxZero.Size.Should().Be(7); // 7
+            ClaimTransaction claimTxOne = GetClaimTransaction(1);
+            claimTxOne.Size.Should().Be(41); // 34 + 7
+            ClaimTransaction claimTxTwo = GetClaimTransaction(2);
+            claimTxTwo.Size.Should().Be(75); // 2*34 + 7
+
+            ClaimTransaction claimTx30 = GetClaimTransaction(30);
+            claimTx30.Size.Should().Be(1027); // 30*34 + 7
+            claimTx30.NetworkFee.Should().Be(Fixed8.Zero);
+            claimTx30.IsLowPriority.Should().Be(true); // by default is Low Priority, but plugin makes it High Priority
+            //uut.IsLowPriority -> cannot inspect because it's private... no problem!
+
+            List<Transaction> TxList = new List<Transaction>();
+            TxList.Insert(0, claimTxZero);
+            TxList.Insert(0, claimTxOne);
+            TxList.Insert(0, claimTxTwo);
+            TxList.Insert(0, claimTx30);
+
+            // ======================== BEGIN TESTS ============================
+
+            // insert invocation transactions (100 of each)
+            for (var i = 0; i < 100; i++)
+            {
+                TxList.Insert(0, MockGenerateInvocationTransaction(Fixed8.One, 50).Object);
+                TxList.Insert(0, MockGenerateInvocationTransaction(Fixed8.Zero, 50).Object);
+            }
+
+            TxList.Count().Should().Be(204); // 100 free + 100 paid + 4 claims
+
+            IEnumerable<Transaction> filteredTxList = uut.FilterForBlock(TxList);
+            //filteredTxList.Count().Should().Be(124); // 20 free + 100 paid + 4 claims
+            filteredTxList.Count().Should().Be(120); // 20 free (including 4 claims) + 100 paid
+
+            // will select 20 low priority (including Claims)
+            var vx = filteredTxList.Where(tx => tx.IsLowPriority == true);
+            vx.Count().Should().Be(20);
+
+            // all Claim Transaction will survive
+            vx = filteredTxList.Where(tx => tx.Type == TransactionType.ClaimTransaction);
+            vx.Count().Should().Be(4);
+
+            // =================================================================
+
+            // insert more invocation transactions (400 of each)
+            for (var i = 0; i < 400; i++)
+            {
+                TxList.Insert(0, MockGenerateInvocationTransaction(Fixed8.One, 50).Object);
+                TxList.Insert(0, MockGenerateInvocationTransaction(Fixed8.Zero, 50).Object);
```

Comment:
It could be an issue Erik (I'll verify updated ordering), if Claim is "HighPriority", it needs to gain from other low priority (even if other is 0.0009). As it is now it's testing only if it gains from Zero fee tx, so there will be no real priority.

Reply:
Ok now, I remake the tests in a way that some Claims will pass and others will not pass.